Source/Applications/SystemCenterNotification/Scripts/TSX/EmailTypes/EmailPage.tsx

```diff
@@ -57,56 +57,71 @@ const EmailPage = (props: IProps) => {
                 ]} />
+            </div>
 
-               
-                {tab == "settings" ? <div className={"tab-pane active"} id="settings">
-                    <GeneralInfo Record={email} />
-                </div> : null}
-                {tab == "template" ? <div className={"tab-pane active"} id="template">
-                    <Template Record={email} />
-                </div> : null}
-                {tab == "dataSources" ? <div className={"tab-pane active"} id="dataSources">
-                    <DataSourceWindow Record={email} />
-                </div> : null}
-                {tab == 'trigger' ? <div className={"tab-pane active"} id="trigger">
-                    <TriggerWindow Record={email} />
-                </div> : null}
-                {tab == 'subscriptions' ? <div className={"tab-pane active"} id="subscriptions">
-                    <Subscriptions Record={email} />
-                </div> : null}
-                <TestEmail show={showTest} OnClose={() => setShowTest(false)} record={email} />
-                <Warning Message={'This will permanently delete this notification and can not be undone.'} Show={showDelete} Title={'Delete ' + (email !== undefined? email.Name : '')}
-                    CallBack={(conf) => { if (conf) dispatch(EmailTypeSlice.DBAction({ verb: 'DELETE', record: email })); }} />
+            <div className="row" style={{ flex: 1, overflow: 'hidden' }}>
+                <div className="col-12" style={{ padding: 0 }}>
+                    <div className="tab-content" style={{ height: '100%' }}>
+                        {tab == "settings" ?
+                            <div className="tab-pane active" style={{ height: 'inherit' }}>
+                                <GeneralInfo Record={email} />
+                            </div>
+                        : null}
+                        {tab == "template" ?
+                            <div className="tab-pane active" style={{ height: 'inherit' }}>
+                                <Template Record={email} />
+                            </div>
+                            : null}
+                        {tab == "dataSources" ?
+                            <div className="tab-pane active" style={{ height: 'inherit' }}>
+                                <DataSourceWindow Record={email} />
+                            </div>
+                        : null}
+                        {tab == "trigger" ?
+                            <div className="tab-pane active" style={{ height: 'inherit' }}>
+                                <TriggerWindow Record={email} />
+                            </div>
+                            : null}
+                        {tab == "subscriptions" ?
+                            <div className="tab-pane active" style={{ height: 'inherit' }}>
+                                <Subscriptions Record={email} />
+                            </div>
+                        : null}
```

Comment:
React tends to play nicer with using <></> over null if you want to return nothing.

Reply:
This is not done this way in many places, don't worry about changing all of them too much, just whenever you see it.